packages/core/src/state-management/state-manager.ts

```diff
@@ -360,18 +354,16 @@ export class StateManager {
       .pipe(
         takeUntil(stopSignal),
         concatMap(async (asr) => {
-          if (!asr.cid.equals(state$.tip) && asr.status != AnchorStatus.ANCHORED) {
-            // We don't want to change a stream's state due to changes to the anchor
-            // status of a commit that is no longer the tip of the stream, so we early return
-            // in most cases when receiving a response to an old anchor request.
-            // The one exception is if the AnchorServiceResponse indicates that the old commit
-            // is now anchored, in which case we still want to try to process the anchor commit
-            // and let the stream's conflict resolution mechanism decide whether or not to update
-            // the stream's state.
-            return
-          }
+          // We don't want to change a stream's state due to changes to the anchor
+          // status of a commit that is no longer the tip of the stream, so we early return
+          // in most cases when receiving a response to an old anchor request.
+          // The one exception is if the AnchorServiceResponse indicates that the old commit
```

Comment:
this isn't the one exception anymore right, that's why you moved the if-statement this used to be contained in?  Can we update this comment to reflect the new reality?

Reply:
In general, the logic of the comment makes sense, and does no harm.

It is an exception, because of the way CAS works _now_. If something changes down the road, the behaviour stated in the comment and in code would still make sense.